### DISCUSSION TITLE
rust-analyzer doesn't recognize codes under feature "ssr"

### DISCUSSION THREAD

**Discussion by AndyJado** [leptos-rs/leptos#2214] [Q&A] [answered]
0. I get `todo_app_sqlite_axum` runing fine in my broswer

1. load the example workspace and compiled many times with attempts `cargo build & cargo leptos build` and `--features "ssr"` stf

2. R-A in my editor complains anyway:

```rust
#[cfg(feature = "ssr")]
use leptos_axum::ResponseOptions;
```

with 

`code is inactive due to #[cfg] directives: feature = "ssr" is disable`

**Comment by gbj** [comment, marked as answer]
You can configure rust-analyzer to use all features for projects, or to use particular features for different projects. How to do that depends on your editor. `rust-analyzer.cargo.allFeatures` is the relevant setting.

**Reply by AndyJado** [reply]
now I get why `rust-analyzer.cargo.features = "all"` doesn't work 🤡

**Comment by caiigee** [comment]
Does anybody know how to get this working on Zed? I initialized a new project with Axum template and `rust-analyzer` is inactive for both `main.rs` and `lib.rs`...

I have tried doing this:
```
"lsp": {
    "rust-analyzer": {
      "cargo": {
        "features": ["ssr", "hydrate"]
      }
    }
  }
```

but with no luck, it just doesn't work... I have also tried `"allFeatures": true` and `"features": all` and they don't work...

I am using Arch Linux (6.10.10-arch1-1) with Zed (0.153.6) and rust-analyzer version is 1.83.0-nightly.
